modules/nlu/src/backend/language-provider.ts

```diff
@@ -93,15 +106,32 @@ export class RemoteLanguageProvider implements LanguageProvider {
       return []
     }
 
-    return this.queryProvider(lang, '/vectorize', { tokens }, 'vectors')
+    const text = tokens.reduce((a, b) => a + b, '')
+    const cacheKey = `${lang}_${encodeURI(text)}`
+
+    if (this._vectorsCache.has(cacheKey)) {
+      return this._vectorsCache.get(cacheKey)
+    }
+
+    const vectors = await this.queryProvider<number[][]>(lang, '/vectorize', { tokens }, 'vectors')
+    this._vectorsCache.set(cacheKey, vectors)
+    return vectors
   }
 
   async tokenize(text: string, lang: string): Promise<string[]> {
     if (!text.length) {
       return []
     }
 
-    return this.queryProvider(lang, '/tokenize', { input: text }, 'tokens')
+    const cacheKey = `${lang}_${encodeURIComponent(text)}`
```

Comment:
You use encodeURI in the previous one and encodeURIComponent here, can you use the same one?